File: ChatGPT/Direct_Attempt/python/output/output_07a.py
# ChatGPT-generated solution will go here.
from itertools import product

def eval_expr(nums, ops):
    result = nums[0]
    for i, op in enumerate(ops):
        if op == '+':
            result += nums[i + 1]
        else:
            result *= nums[i + 1]
    return result

def main():
    total = 0
    with open('input.txt') as f:
        for line in f:
            if not line.strip():
                continue
            left, right = line.strip().split(':')
            target = int(left.strip())
            nums = list(map(int, right.strip().split()))
            n = len(nums)
            for ops in product(['+', '*'], repeat=n-1):
                if eval_expr(nums, ops) == target:
                    total += target
                    break
    print(total)

if __name__ == '__main__':
    main()

File: ChatGPT/Direct_Attempt/c/output/output_07a.c
// ChatGPT-generated solution will go here.
#include <stdio.h>
#include <stdlib.h>
#include <string.h>

#define MAX_NUMBERS 10

int evaluate(int numbers[], int n, char operators[]) {
    int result = numbers[0];
    for (int i = 0; i < n - 1; i++) {
        if (operators[i] == '+') {
            result += numbers[i + 1];
        } else if (operators[i] == '*') {
            result *= numbers[i + 1];
        }
    }
    return result;
}

int is_valid_equation(int target, int numbers[], int n) {
    int total_ways = 0;
    char operators[] = {'+', '*'};
    int operator_combinations = 1 << (n - 1);  // 2^(n-1) combinations of + and *
    
    for (int mask = 0; mask < operator_combinations; mask++) {
        char ops[MAX_NUMBERS];
        for (int i = 0; i < n - 1; i++) {
            ops[i] = (mask & (1 << i)) ? '*' : '+';
        }

        if (evaluate(numbers, n, ops) == target) {
            total_ways++;
        }
    }
    return total_ways > 0;
}

int main() {
    FILE *f = fopen("input.txt", "r");
    char line[100];
    int total_sum = 0;

    while (fgets(line, sizeof(line), f)) {
        int target, n = 0, numbers[MAX_NUMBERS];
        sscanf(line, "%d:", &target);

        char *token = strtok(line + strlen(line) - strlen(strtok(line, ":")), " ");
        while (token) {
            numbers[n++] = atoi(token);
            token = strtok(NULL, " ");
        }

        if (is_valid_equation(target, numbers, n)) {
            total_sum += target;
        }
    }

    fclose(f);
    printf("%d\n", total_sum);
    return 0;
}